app/models/concerns/events/courses/state.rb

```diff
@@ -49,7 +49,7 @@ module Events::Courses::State
   def send_canceled_email
     return if canceled_reason.nil?
 
-    all_participants.each do |participation|
+    all_participants.where.not(previous_state: :canceled).find_each do |participation|
```

Comment:
Im `previous_state` steht nur 'canceled' drin, wenn der participation state auf etwas anders gewechselt wurde, also nicht mehr 'canceled' ist (z.B. 'annulled').

Ist das korrekt in diesem Kontext?

Reply:
-> ist korrekt, verifiziert mit regression spec